src/Database/Mysql/Driver/Mysql.php

```diff
@@ -54,15 +59,46 @@ public function update(Tenant $tenant): bool
             return false;
         }
 
-        return $this->process($tenant, [
-            'user' => "RENAME USER `{$config['oldUsername']}`@'{$config['host']}' TO `{$config['username']}`@'{$config['host']}'",
-        ]);
+        if (isset($config['allowedHost'])) {
+            $config['host'] = $config['allowedHost'];
+        }
+
+        $tempTenant = $tenant;
+        $tempTenant->{$tempTenant->getTenantKeyName()} = $tenant->getOriginal($tenant->getTenantKeyName());
+        $originalTenant = Tenancy::getTenant();
+
+        Tenancy::setTenant($tempTenant);
+        $connection = Tenancy::getTenantConnection();
+        $tables = $connection->select('SHOW TABLES');
```

Comment:
Wouldn't the database schema class of the connection have knowledge of all tables already?

Reply:
Yeah thought so too, however, when I was working on it, I figured out it didn't.
I will do some more digging into this today.

Reply:
Had a look, what we could do instead is require doctrine/dbal and use that functionality from the laravel connection. Would prevent the raw select and replace it with:
`$connection->getDoctrineSchemaManager()->listTableNames()`

Reply:
I'm definitely in favor of `doctrine/dbal` for this.  Otherwise you risk users trying to use some DBMS where `SHOW TABLES` isn't valid

Reply:
Really appreciate the feedback @fletch3555.
As @luceos also upvoted, decided to install dbal and changed the code.

Reply:
Thinking about it now, this is the mysql driver, so other DBMS shouldn't matter. But either way, I think it's cleaner with dbal